src/main/scala/org/broadinstitute/dsde/workbench/leonardo/service/LeonardoService.scala

```diff
@@ -229,21 +247,27 @@ class LeonardoService(protected val dataprocConfig: DataprocConfig,
     }
   }
 
-  private[service] def generateServiceAccountKey(googleProject: GoogleProject, serviceAccountEmail: WorkbenchEmail): Future[Option[ServiceAccountKey]] = {
-    // Only generate a key if NOT creating the cluster as the pet service account.
-    // If the pet service account is used to create a cluster, its credentials are on the metadata
-    // server and we don't need to propagate a key file.
-    if (!dataprocConfig.createClusterAsPetServiceAccount) {
-      googleIamDAO.createServiceAccountKey(dataprocConfig.leoGoogleProject, serviceAccountEmail).map(Option(_))
-    } else Future.successful(None)
+  private[service] def generateServiceAccountKey(googleProject: GoogleProject, serviceAccountOpt: Option[WorkbenchEmail]): Future[Option[ServiceAccountKey]] = {
+    serviceAccountOpt.traverse { serviceAccountEmail =>
+      googleIamDAO.createServiceAccountKey(dataprocConfig.leoGoogleProject, serviceAccountEmail)
+    }
+  }
+
+  private[service] def removeServiceAccountKey(googleProject: GoogleProject, clusterName: ClusterName, serviceAccountOpt: Option[WorkbenchEmail]): Future[Unit] = {
+    // Delete the service account key in Google, if present
+    val tea = for {
+      key <- OptionT(dbRef.inTransaction { _.clusterQuery.getServiceAccountKeyId(googleProject, clusterName) })
+      serviceAccountEmail <- OptionT.fromOption[Future](serviceAccountOpt)
+      _ <- OptionT.liftF(googleIamDAO.removeServiceAccountKey(dataprocConfig.leoGoogleProject, serviceAccountEmail, key))
+    } yield ()
+
+    tea.value.void
```

Comment:
suck it, baby shoes

Reply:
I don't get this reference but I appreciate the comment!

Reply:
This is a better short story than "for sale, baby shoes, never worn"